metadata.go

```diff
@@ -331,6 +331,9 @@ func getKeyspaceMetadata(
 		`,
 		keyspaceName,
 	)
+	// Set a routing key to avoid GetRoutingKey from computing the routing key
+	// TODO use a separate connection (pool) for system keyspace queries.
+	query.RoutingKey([]byte{})
```

Comment:
It's only necessary here since these methods are in the call path of GetRoutingKey.

More specifically, each of these get***Metadata functions execute a query using the connection pool, which may be implemented with token-awareness, which will in turn call GetRoutingKey, and therefore would infinitely loop recursively; except that by overriding the routing key here in this line, this short circuits the behavior of GetRoutingKey to return the override value immediately and avoid the recursion.

The ideal situation, as indicated by the TODO is that these functions execute their queries on an isolated connection (e.g. control connection) that wouldn't ever call GetRoutingKey.